svg/lesson3/tutorial.md

```diff
@@ -0,0 +1,608 @@
+                <rect x="150" y="150" width="40" height="40" stroke="black" stroke-width="3" fill="red" />
+                <rect x="100" y="100" width="60" height="60" stroke="black" stroke-width="3" fill="green" />
+            </g>
+        </svg>
+    </body>
+</html>
+```
+
+<svg id="example1" width="200px" height="200px">
+    <g>
+        <rect x="50" y="50" width="120" height="80" stroke="black" stroke-width="3" fill="red" />
+        <rect x="150" y="150" width="40" height="40" stroke="black" stroke-width="3" fill="red" />
+        <rect x="100" y="100" width="60" height="60" stroke="black" stroke-width="3" fill="green" />
+    </g>
+</svg>
+
+What it allows you to do though is to define style attributes for the whole
+group that can be overriden for individual shapes. For instance, in the
+example below, the `stroke`, `stroke-width` and `fill` properties are now
+defined on the group rather than the individual rectangles and the third
+rectangle overrides the `fill` property:
+
+```html
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>SVG Tutorial</title>
+    </head>
+    <body>
+        <svg width="200px" height="200px">
+            <g stroke="black" stroke-width="3" fill="red">
+                <rect x="50" y="50" width="120" height="80" />
+                <rect x="150" y="150" width="40" height="40" />
+                <rect x="100" y="100" width="60" height="60" fill="green" />
+            </g>
+        </svg>
+    </body>
+</html>
+```
+
+<svg id="example2" width="200px" height="200px">
+    <g stroke="black" stroke-width="3" fill="red">
+        <rect x="50" y="50" width="120" height="80" />
+        <rect x="150" y="150" width="40" height="40" />
+        <rect x="100" y="100" width="60" height="60" fill="green" />
+    </g>
+</svg>
+
+What can be done with SVG attributes can also be done with CSS:
+
+```html
+<!DOCTYPE html>
+<html>
+    <head>
+        <title>SVG Tutorial</title>
+        <style>
+        g {
+            stroke: black;
+            stroke-width: 3;
+            fill: red;
+        }
+        rect.accent {
+            fill: green;
+        }
+        </style>
+    </head>
+    <body>
+        <svg width="200px" height="200px">
+            <g>
+                <rect x="50" y="50" width="120" height="80" />
+                <rect x="150" y="150" width="40" height="40" />
+                <rect x="100" y="100" width="60" height="60" class="accent" />
+            </g>
+        </svg>
+    </body>
+</html>
+```
+
+<svg id="example3" width="200px" height="200px">
+    <style>
+    #example3 g {
+        stroke: black;
+        stroke-width: 3;
+        fill: red;
+    }
+    #example3 rect.accent {
+        fill: green;
+    }
+    </style>
+    <g>
+        <rect x="50" y="50" width="120" height="80" />
+        <rect x="150" y="150" width="40" height="40" />
+        <rect x="100" y="100" width="60" height="60" class="accent" />
+    </g>
+</svg>
+
+## Transforms
```

Comment:
we should make clear that you can also apply transforms in CSS.

Reply:
Yes, good point, I'll mention that.

Reply:
I added a note on CSS transforms and a link to the Mozilla documentation at the bottom.